### ISSUE TITLE
Scheduled messages

### ISSUE THREAD

**Issue by serrq** [simplex-chat/simplex-chat#1549]
Cases like "I have to send this message at noon, but I’m in a meeting and I might forget of it" or "It’s midnight, I don’t want to disturb now" can finally have a dedicated solution.

Sliding sidely the send button (instead of pressing) you can select the date and time of postponed sending.

Of course you can cancel the sending as long as the time allows.

**Comment by x80486** [comment]
Isn't this more appropriate for something like e-mail? The nature of messaging applications are orthogonal to this concept...but still, you can implement such thing — though I highly doubt it will be used a lot.

On the other hand, if the messaging application is used more towards work-related things, the landscape changes a little bit.

**Comment by SamiLehtinen** [comment]
Sometimes message scheduling could have privacy implications and benefits. As example it could be used to obscure senders timezone information etc.

**Comment by blakeashleyjr** [comment]
Correlation attacks are a common way of associating someone with their online activity. For example, if someone only sends messages during a certain 14-16 hour window, that is likely daytime for them.

Or if you watch someone for a while, you can determine when they are using their devices vs when messages go out.

The ability to schedule messages would defeat this type of attack and invasion of privacy!

**Comment by greensheeps** [comment]
Signal has since added this. Scheduled messages only get sent if the senders device is online, the other obvious easy option is sending the message to the other client(s) immediately but not displaying it until it's supposed to be delivered, modified client could bypass this. Possibly could be sent to servers and they forward it on when it's supposed to arrive?

**Comment by gUser5689** [comment]
First I thought this was an implemented solution but obviously it is a suggested implementation. I think "press and hold" is better though, to get to the "schedule send" option.


> Cases like "I have to send this message at noon, but I’m in a meeting and I might forget of it" or "It’s midnight, I don’t want to disturb now" can finally have a dedicated solution.
> 
> Sliding sidely the send button (instead of pressing) you can select the date and time of postponed sending.
> 
> Of course you can cancel the sending as long as the time allows.

**Comment by greensheeps** [comment]
> First I thought this was an implemented solution but obviously it is a suggested implementation. I think "press and hold" is better though, to get to the "schedule send" option.
> 
> > Cases like "I have to send this message at noon, but I’m in a meeting and I might forget of it" or "It’s midnight, I don’t want to disturb now" can finally have a dedicated solution.
> > Sliding sidely the send button (instead of pressing) you can select the date and time of postponed sending.
> > Of course you can cancel the sending as long as the time allows.

Yes, I agree. Press and hold already shows buttons for sending live message and per-message disappearing message. It makes sense that scheduled messages would be there as well.

**Comment by gUser5689** [comment]
But I think "Press and hold" should bring up a menu with different alternatives such as "schedule messages", etc. "Press and hold" on mobile phones are often equivalent with right click on a computer which usually brings up some kind of menu options.

> >  think "press and hold" is better
> 
> Deliberately I left this action "empty" for future features.
> 
> For example: payments.
